src/Database/Adapter/Mongo.php

```
@@ -1844,33 +1856,51 @@ protected function getInternalKeyForAttribute(string $attribute): string
-    public function find(Document $collection, array $queries = [], ?int $limit = 25, ?int $offset = null, array $orderAttributes = [], array $orderTypes = [], array $cursor = [], string $cursorDirection = Database::CURSOR_AFTER, string $forPermission = Database::PERMISSION_READ): array
-    {
+    public function find(
+        QueryContext $context,
+        ?int $limit = 25,
+        ?int $offset = null,
+        array $cursor = [],
+        string $cursorDirection = Database::CURSOR_AFTER,
+        string $forPermission = Database::PERMISSION_READ,
+        array $selects = [],
+        array $filters = [],
+        array $joins = [],
```

PHPMD (2.15.0)

Avoid unused parameters such as '$joins'. (undefined)

(UnusedFormalParameter)